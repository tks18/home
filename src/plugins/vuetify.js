import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import minifyTheme from 'minify-css-string'

Vue.use(Vuetify);

export default new Vuetify({
  theme: {
    themes: {
      light: {
        primary: '#F44336',
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107'
      },
    },
    options: {
      customProperties: true,
      minifyTheme: minifyTheme,
      themeCache: {
        get: key => localStorage.getItem(key),
        set: (key, value) => localStorage.setItem(key, value),
      },
    },
  },
});
