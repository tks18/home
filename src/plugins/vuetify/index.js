import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import minifyTheme from 'minify-css-string';

Vue.use(Vuetify);

let themecache = JSON.parse(localStorage.getItem('themecache'));

export default new Vuetify({
  theme: {
    dark: themecache == null ? false : themecache.dark,
    themes: {
      light: {
        primary: themecache == null ? '#F44336' : themecache.theme.accent,
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107',
      },
      dark: {
        primary: themecache == null ? '#F44336' : themecache.theme.accent,
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107',
      },
    },
    options: {
      minifyTheme: minifyTheme,
      variations: false,
    },
  },
});
