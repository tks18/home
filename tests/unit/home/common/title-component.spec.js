import Vuetify from 'vuetify';
import { vuetifyOptions } from '@p/vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import title_component from '@v/home/components/common/title-component';

describe('Common Home Title Component.vue', () => {
  const localVue = createLocalVue();
  let vuetify;
  let propsData;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify(vuetifyOptions);
    propsData = {
      id: 'testing-id',
      title: 'Testing Title',
    };
    wrapper = mount(title_component, {
      localVue,
      vuetify,
      propsData,
    });
  });
  it('Checks Vue instance and Matches with Snapshot', () => {
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Checks the Element id with the passed id', () => {
    const div = wrapper.find('div');
    expect(div.attributes('id')).toBe(`home-${propsData.id}`);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Checks Title text with the Props Sent', () => {
    const title = wrapper.find(`#home-${propsData.id}`);
    expect(title.exists()).toBeTruthy();
    expect(title.text()).toBe(propsData.title);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
