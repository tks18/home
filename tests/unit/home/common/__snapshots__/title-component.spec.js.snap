// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Home Title Component.vue Checks Vue instance and Matches with Snapshot 1`] = `
<div id="home-testing-id" class="clip-text-back text-h5  ml-6  non-touch point-cursor text-capitalize underhover-dark">
  Testing Title <i aria-hidden="true" class="v-icon notranslate mdi mdi-arrow-right-circle theme--light"></i></div>
`;
